File: Demo2/app/src/main/java/com/example/demo2/ListFragment.java
package com.example.demo2;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;

public class ListFragment extends Fragment {

    @NonNull
    public static ListFragment newInstance() { return new ListFragment(); }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_list, container, false);
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        ArrayList<Integer> data = new ArrayList<>();
        for (int i = 42; i > 0; i -= 1) {
            data.add(i);
        }

        RecyclerView recycler = getView().findViewById(R.id.recycler);
        RecyclerView.LayoutManager manager = new LinearLayoutManager(getContext());
        recycler.setLayoutManager(manager);
        RecyclerView.Adapter adapter = new Adapter(getContext(), data);
        recycler.setAdapter(adapter);
    }
}

File: Demo2-FragmentsAndRecyclerView-Kotlin/app/src/main/java/com/example/fragmentandrecyclerviewdemo_kotlin/ui/dashboard/ListFragment.kt
package com.example.fragmentandrecyclerviewdemo_kotlin.ui.dashboard

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.fragmentandrecyclerviewdemo_kotlin.R

class ListFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val root = inflater.inflate(R.layout.fragment_list, container, false)
        val recycler: RecyclerView = root.findViewById(R.id.number_list)
        recycler.layoutManager = LinearLayoutManager(context)
        return root
    }
}